From: =?utf-8?q?Jonas_Dre=C3=9Fler?= <verdre@v0yd.nl>
Date: Mon, 8 Mar 2021 13:55:01 +0100
Subject: Revert "backends: Use also a native cursor renderer for tablets"

With commit c98575344295b715a0301c7319195b2724dc2bde the support for
multiple hardware cursors broke, but those were never properly supported
anyway as we usually assume there's only one hardware cursor around.

With the introduction of the KMS thread in the future, we'll only have
one KMS cursor that gets updated directly from the input thread. So
apart from the fact that it never really makes sense to have two cursors
visible, in this new model having multiple cursors won't work anyway.

So make the cursor we show for stylii a software cursor again.
Eventually the plan is to make the input device that's driving the KMS
cursor interchangeable, so that we can always use hardware cursors.

This reverts commit 165b7369c88644867a7c0c2791c48826240b63b5.

Fixes https://gitlab.gnome.org/GNOME/mutter/-/issues/1645

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1758>
---
 src/backends/native/meta-seat-native.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/backends/native/meta-seat-native.c b/src/backends/native/meta-seat-native.c
index 9b93193..a84f8e4 100644
--- a/src/backends/native/meta-seat-native.c
+++ b/src/backends/native/meta-seat-native.c
@@ -61,7 +61,7 @@ meta_seat_native_handle_event_post (ClutterSeat        *seat,
 
   if (event_type == CLUTTER_PROXIMITY_IN)
     {
-      MetaCursorRendererNative *cursor_renderer_native;
+      MetaCursorRenderer *cursor_renderer;
 
       if (!seat_native->tablet_cursors)
         {
@@ -69,10 +69,9 @@ meta_seat_native_handle_event_post (ClutterSeat        *seat,
                                                                g_object_unref);
         }
 
-      cursor_renderer_native =
-        meta_cursor_renderer_native_new (meta_get_backend (), device);
+      cursor_renderer = meta_cursor_renderer_new (meta_get_backend (), device);
       g_hash_table_insert (seat_native->tablet_cursors,
-                           device, cursor_renderer_native);
+                           device, cursor_renderer);
       return TRUE;
     }
   else if (event_type == CLUTTER_PROXIMITY_OUT)
